lib/internal/encoding.js

```diff
@@ -419,10 +421,13 @@ function makeTextDecoderICU() {
       this[kFatal] = Boolean(options?.fatal);
       // Only support fast path for UTF-8.
       this[kUTF8FastPath] = enc === 'utf-8';
+      this[kLatin1FastPath] = enc === 'windows-1252';
       this[kHandle] = undefined;
 
-      if (!this[kUTF8FastPath]) {
-        this.#prepareConverter();
+      if (this[kUTF8FastPath]) {
+        decodeUTF8(this.input, this[kIgnoreBOM], this[kFatal]);
```

Comment:
Please revert this.

Reply:
how should I update line 425
 ```
       if (!this[kUTF8FastPath] || !this[kLatin1FastPath]) {
        this.#prepareConverter();
      }
 ```
 Is that true?